spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/AggregateReader.java

```diff
@@ -89,6 +96,35 @@ public Iterable<T> findAllById(Iterable<?> ids) {
 		return jdbcTemplate.query(sqlGenerator.findAllById(), Map.of("ids", convertedIds), this::extractAll);
 	}
 
+	public Iterable<T> findAllBy(Query query) {
```

Comment:
`findAll` and others should be rewritten to use this method. I'm also not sure why we always return `Iterable`.

Reply:
We should return List types in the Template API at least. Not everything needs to be used in the repository although almost all modules return List from their `Simple…Repository`.

Reply:
Then let's just go all the way to `SimpleRepository`

Reply:
I filed #1623 to keep track of that change.